test/unit/specs/mount/options/mocks.spec.js

```diff
@@ -84,4 +84,65 @@ describe('mount.mocks', () => {
     const freshWrapper = mount(Component)
     expect(typeof freshWrapper.vm.$store).to.equal('undefined')
   })
+
+  it('replaces lifecycle hooks, but not mixin hooks with mocks', (done) => {
```

Comment:
Could you split this into two tests, one for the lifecycle hooks, and one that checks mixin hooks aren't replaced